Add another GRMHD conservative to primitive scheme

kidder opened the pull request:
## Proposed changes

<!--
At a high level, describe what this PR does.
-->

### Types of changes:

- [ ] Bugfix
- [x] New feature

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

kidder committed:
Add another GRMHD primitive to conservative scheme

fmahebert commented on src/Evolution/Systems/GrMhd/ValenciaDivClean/PalenzuelaEtAl.cpp:
```diff
@@ -0,0 +1,173 @@
+  const double s_;
+  const double t_squared_;
+  const double rest_mass_density_times_lorentz_factor_;
+    const EquationsOfState::EquationOfState<true, ThermodynamicDim>&
```
Indentation problem here

fmahebert commented on src/Evolution/Systems/GrMhd/ValenciaDivClean/PalenzuelaEtAl.hpp:
```diff
@@ -0,0 +1,71 @@
+ * the scheme of [Palenzuela et al, Phys. Rev. D 92, 044045
+ * (2015)](https://doi.org/10.1103/PhysRevD.92.044045).
+ *
+ * In the Palenzuela paper, `total_energy_density` is \f$D (1+q)\f$,
```
Here I am not totally clear whether you are defining the code's input variables in terms of the variables in the Palenzuela paper, or in terms of "our" variables.

kidder replied:
I state "In the Palenzuela paper...", so what would make it more clear?

fmahebert replied:
I want to know if the "rhs" (in "X is Y" statements) D is the same D as what we use, or different.

If it is the same, then I am not sure what the statement  "In the Palenzuela paper..." is trying to tell me (because it sounds like it's telling me it's different). In this scenario, I would find it clearer to remove this statement.

If it is different, then I would either rewrite the equations in terms of our variables, or make explicit (as you have below for h) that D is not our D, and give the relation between the two.

fmahebert commented on src/Evolution/Systems/GrMhd/ValenciaDivClean/PalenzuelaEtAl.hpp:
```diff
@@ -0,0 +1,71 @@
+ * q = & \frac{{\tilde \tau}}{{\tilde D}} \\
+ * r = & \frac{\gamma^{mn} {\tilde S}_m {\tilde S}_n}{{\tilde D}^2} \\
+ * t^2 = & \frac{({\tilde B}^m {\tilde S}_m)^2}{{\tilde D}^3 \sqrt{\gamma}} \\
+ * s = & \frac{\gamma_{mn} {\tilde B}^m {\tilde B}^n}{{\tilde D}\sqrt{\gamma}}
```
As far as I can tell, q,r,t,s are not visible to the user of this class. Should this paragraph be made a Doxygen detail, or perhaps even moved to the .cpp?

fmahebert replied:
Oh, I see now that these are likely intended to define the q,r,t,s in the other doxygen comment. Could the two comments be combined into one?

fmahebert reviewed (commented):
Small comments on dox, can be squashed directly.

nilsdeppe commented on src/Evolution/Systems/GrMhd/ValenciaDivClean/PalenzuelaEtAl.hpp:
```diff
@@ -0,0 +1,71 @@
+      const EquationsOfState::EquationOfState<true, ThermodynamicDim>&
+          equation_of_state) noexcept;
+
+  static const std::string name() noexcept { return "Newman Hamlin"; }
```
Oops... `Newman Hamlin` -> ...

nilsdeppe commented on src/Evolution/Systems/GrMhd/ValenciaDivClean/PalenzuelaEtAl.cpp:
```diff
@@ -0,0 +1,173 @@
+
+namespace {
+template <size_t ThermodynamicDim>
+class FunctionOfX {
```
I would've found a short comment `// Note: the variable x is defined in the documentation` useful since `x` is usually a spatial coordinate.

nilsdeppe commented on src/Evolution/Systems/GrMhd/ValenciaDivClean/PalenzuelaEtAl.hpp:
```diff
@@ -0,0 +1,71 @@
+ * enthalpy times the rest mass density.
+ *
+ */
+
```
These comments should be joined.

nilsdeppe reviewed (changes requested):
I'm happy for you to squash my comments

kidder committed:
Remove exp10

exp10 is not in the standard library, and fails with AppleClang
Also compute the size of tuples instead of hard-coding them.

nilsdeppe commented:
@wthrowe could you please look at this one when you get a chance? Thanks!

nilsdeppe commented:
Travis timeouts when building last executables
